otter_welcome_buddy/database/db_helpers.py

```diff
@@ -0,0 +1,172 @@
+from collections import namedtuple
+from collections.abc import Callable
+from pathlib import Path
+from sqlite3 import Cursor
+from typing import Any
+
+from otter_welcome_buddy.common.constants import DATA_FILE_PATH
+from otter_welcome_buddy.database.dbconn import DbContext
+
+
+def namedtuple_factory(cursor: Cursor, row: tuple) -> tuple:
+    """Returns sqlite rows as named tuples"""
+    fields: list[str] = [col[0] for col in cursor.description if col[0].isidentifier()]
+    named_row = namedtuple("Row", fields)  # type: ignore
+    return named_row(*row)
+
+
+def dict_factory(cursor: Cursor, row: tuple) -> dict:
+    """Returns sqlite rows as dict"""
+    col_names = [col[0] for col in cursor.description]
+    return dict(zip(col_names, row))
+
+
+def insert_one(table: str, columns: list[str], values: tuple) -> int:
+    """Generic insert of one value into a table
+    specifying the columns and values desired"""
+    row_count = 0
+    n_columns = len(columns)
+    query = f"""
+        INSERT OR REPLACE
+        INTO {table}
+        ({", ".join(columns)})
+        VALUES ({", ".join(["?"] * n_columns)})
+    """
+
+    with DbContext(Path(DATA_FILE_PATH)) as conn:
+        if conn.connection is None or conn.cursor is None:
+            raise ConnectionError("The connection to the database is not open")
+        row_count = conn.cursor.execute(query, values).rowcount
+        conn.connection.commit()
+
+    return row_count
+
+
+def insert_many(table: str, columns: list[str], values: list[tuple]) -> int:
+    """Generic insert of multiple values into a table
+    specifying the columns and values desired"""
+    row_count = 0
+    n_columns = len(columns)
+    query = f"""
+        INSERT OR REPLACE
+        INTO {table}
+        ({", ".join(columns)})
+        VALUES ({", ".join(["?"] * n_columns)})
+    """
+
+    with DbContext(Path(DATA_FILE_PATH)) as conn:
+        if conn.connection is None or conn.cursor is None:
+            raise ConnectionError("The connection to the database is not open")
+        row_count = conn.cursor.executemany(query, values).rowcount
+        conn.connection.commit()
+
+    return row_count
+
+
+def update(
```

Comment:
Consider to name all the methods with a more explicit name

Reply:
This is inside the db_helpers, and is an update, do u think is not explicit enough?